### ISSUE TITLE
Logging from Swift leaks Strings

### ISSUE THREAD

**Issue by c0diq** [CocoaLumberjack/CocoaLumberjack#774]
### New Issue Checklist
- [X] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
- [X] I have read the [Documentation](https://github.com/CocoaLumberjack/CocoaLumberjack#documentation) or [CocoaDocs](cocoadocs.org/docsets/CocoaLumberjack)
- [X] I have searched for a similar issue in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/issues) and found none
### Issue Info

 Platform Name           | ios
 Platform Version        | 9.0
 CocoaLumberjack Version | 2.4.0
 Integration Method      | manually
 Xcode Version           | Xcode 8.0
 Repro rate              | all the time (100%)
### Issue Description and Steps

When logging from Swift using CocoaLumberjackSwift framework, the strings are not removed from the binary when the level indicates it should.
This is because Swift compiler cannot optimize out code like it does with Objective-C when using static const log level inside macros.
As such, the string "Hello World" can be found in the binary when invoking

```
DDLogDebug(message: "Hello World", level: .DDLogLevelInfo)
```

Expected behavior is that it should be compiled out.

Interestingly, this code properly removes the string from the binary:

```
if DDLogLevel.Info.rawValue & DDLogFlag.Debug.rawValue != 0 {
      let logMessage = DDLogMessage(message: "Hello World", level: .Info, flag: .Debug, context: 0, file: "foo", function: "bar", line: #line, tag: nil, options: [.CopyFile, .CopyFunction], timestamp: nil)
      DDLog.sharedInstance().log(true, message: logMessage)
}
```

**Comment by diederich** [comment]
Is this for Debug or Release builds? What are the optimisation settings?

**Comment by c0diq** [comment]
Release builds. Swift Fast Single-File Optimization [-O].

**Comment by bpoplauschi** [comment]
This is beyond my knowledge. If anyone knows how to fix this, please let us know.
@rivera-ernesto @MaddTheSane

**Comment by ffried** [comment]
I've just reproduced this with Xcode 10 using Swift 4.2 (with an iOS 12 target).

I also added inlining attributes to the Swift DDLog... functions. And in the binary these functions actually were inlined. However the strings still were in the binary, too.

The problem is, that `defaultLogLevel` is a normal (global) variable. It could change at any time. So the compiler can't remove the code inside the `if`:
https://github.com/CocoaLumberjack/CocoaLumberjack/blob/9ac52fd571db600f365522d93548fa397ad66de6/Classes/CocoaLumberjack.swift#L56

That's also visible in the binary. While the `flag` is static here, the `level` is a reference to the global variable.
